docs/guides/publish_fetch_packages.md

```diff
@@ -0,0 +1,177 @@
+</figure>
+
+!!! warning "Important"
+    Do not confuse the **local registry** with a local **workspace folder**. The local registry is simply a repository to store finalized packages, mimicking the role of a remote registry on the local machine.
+
+    For example, you can develop a component package directly in the local registry, or you can develop it in an agent within a workspace folder, and push the component later in the local registry. Then, you can retrieve that component from the registry for use in another agent.
+
+    **Regardless of the methodology that you choose to develop your packages, it is important to have a clear separation of concerns in mind so that they can be reused in other agent projects.**
+
+## How to tell the framework what registry to use
+The `push`, `add`, `publish` and `fetch` commands use, by default, the registry specified when the framework was initiated (command `autonomy init`). See for example the [set up guide](./set_up#set-up), where we initialized the framework to use the default remote [IPFS](https://ipfs.io) registry.
+
+Additionally, the framework configuration can be overridden per command by using the flags `--local` or `--remote` in any of the commands  `push`, `add`, `publish` or `fetch`.
+
+
+## Push and add components
+This section assumes that you have a newly created component, e.g., an {{fsm_app}} skill, within an agent folder. It is out of the scope of this guide how to create agent components. You can review the [guide to create an {{fsm_app}}](./create_fsm_app.md), or the {{open_aea_doc}}. Your component should be located within an agent folder (i.e., a folder containing the file `aea-config.yaml`) in
+```
+<agent_folder>/vendor/<author_name>/<component_type>/<package_name>.
+```
+Here, `<component_type>` is one of `connections`, `contracts`, `protocols` or `skills`.
+
+### Push a component on a registry
+
+1. Ensure that the framework has been [initialized with the desired registry](#how-to-tell-the-framework-what-registry-to-use) (local or remote).
+
+2. Within the agent folder, run the command
+    ```bash
+    autonomy push <component_type> ./vendor/<author_name>/<component_type>/<component_name>
+    ```
+3. You should see a message
+    ```bash
+    Pushed component with:
+    	PublicId: <author_name>/<package_name>:<version>
+    	Package hash: <hash>
+    ```
+
+If you are using the default remote IPFS registry (`/dns/registry.autonolas.tech/tcp/443/https`), you can check that the service has been successfully pushed by accessing the gateway https://gateway.autonolas.tech/ipfs/`<hash>`, by substituting the appropriate `<hash>` value.
+
+### Add a component from a registry
+
+1. Ensure that the framework has been [initialized with the desired registry](#how-to-tell-the-framework-what-registry-to-use) (local or remote).
+
+2. Within the agent folder, run the command
+    ```bash
+    autonomy add <component_type> <author_name>/<package_name>:<version>:<hash>
```

Comment:
I think we don't recommend developing agents outside the local repository anymore, but I'm not 100% sure.

Reply:
Can we clarify this with @angrybayblade or @DavidMinarsch ?

Reply:
Yes we recommend developing in the local package registry. You could simply add a sentence explaining what's recommended.